packages/poseidon-cipher/src/poseidon-cipher.circom

```diff
@@ -107,13 +108,9 @@ template PoseidonDecryptIterations(length) {
     signal output decrypted[decryptedLength];
     signal output decryptedLast;
 
-    var two128 = 2 ** 128;
-
     // nonce must be < 2^128
-    component lt = LessThan(252);
-    lt.in[0] <== nonce;
-    lt.in[1] <== two128;
-    lt.out === 1;
+    component n2b = Num2Bits(128);
+    n2b.in <== nonce;
```

Comment:
When nonce > 2^252, the original LessThan(252) would not work. Changing to use Num2Bits fixes that.

Rationale:
```
// Inside LessThan(252):
n2b.in <== nonce + 2^252 - 2^128
```
Due to field arithmetic modulo p (where p ≈ 2^254):
nonce + 2^252 wraps around to a smaller value
The wrapped result minus 2^128 could be a valid number for Num2Bits
Num2Bits would accept this wrapped value
The comparison result would be incorrect